contribs/src/main/java/com/netflix/conductor/contribs/kafka/KafkaPublishTask.java

```diff
@@ -0,0 +1,271 @@
+		om.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+		return om;
+	}
+
+	@Override
+	public void start(Workflow workflow, Task task, WorkflowExecutor executor) {
+
+		long taskStartMillis = Instant.now().toEpochMilli();
+		task.setWorkerId(config.getServerId());
+		Object request = task.getInputData().get(requestParameter);
+
+		if (Objects.isNull(request)) {
+			task.setReasonForIncompletion(MISSING_REQUEST);
+			task.setStatus(Task.Status.FAILED);
+			return;
+		}
+
+		KafkaPublishTask.Input input = om.convertValue(request, KafkaPublishTask.Input.class);
+
+		if (StringUtils.isBlank(input.getBootStrapServers())) {
+			task.setReasonForIncompletion(MISSING_BOOT_STRAP_SERVERS);
+			task.setStatus(Task.Status.FAILED);
+			return;
+		}
+
+		if (StringUtils.isBlank(input.getTopic())) {
+			task.setReasonForIncompletion(MISSING_KAFKA_TOPIC);
+			task.setStatus(Task.Status.FAILED);
+			return;
+		}
+
+		if (Objects.isNull(input.getValue())) {
+			task.setReasonForIncompletion(MISSING_KAFKA_BODY);
+			task.setStatus(Task.Status.FAILED);
+			return;
+		}
+
+		try {
+			Future<RecordMetadata> recordMetaDataFuture = kafkaPublish(input);
+			try {
+				recordMetaDataFuture.get();
+				task.setStatus(Task.Status.COMPLETED);
+				long timeTakenToCompleteTask = Instant.now().toEpochMilli() - taskStartMillis;
+				logger.info("Published message {}, Time taken {}", input, timeTakenToCompleteTask);
+
+			} catch (ExecutionException ec) {
+				logger.error("Failed to invoke kafka task - execution exception {}", ec);
+				task.setStatus(Task.Status.FAILED);
+				task.setReasonForIncompletion(FAILED_TO_INVOKE + ec.getMessage());
+			}
+		} catch (Exception e) {
+			logger.error(String.format("Failed to invoke kafka task for input {} - unknown exception: {}", input), e);
+			task.setStatus(Task.Status.FAILED);
+			task.setReasonForIncompletion(FAILED_TO_INVOKE + e.getMessage());
+		}
+	}
+
+	/**
+	 * @param input Kafka Request
+	 * @return Future for execution.
+	 */
+	 private Future<RecordMetadata> kafkaPublish(KafkaPublishTask.Input input) throws Exception {
+
+		long startPublishingEpochMillis = Instant.now().toEpochMilli();
+
+		Producer producer = producerManager.getProducer(input);
+
+		long timeTakenToCreateProducer = Instant.now().toEpochMilli() - startPublishingEpochMillis;
+
+		logger.info("Time taken getting producer {}", timeTakenToCreateProducer);
+
+		Object key = getKey(input);
+
+		Iterable<Header> headers = input.getHeaders().entrySet().stream()
+				.map(header -> new RecordHeader(header.getKey(), String.valueOf(header.getValue()).getBytes()))
+				.collect(Collectors.toList());
+		ProducerRecord rec = new ProducerRecord(input.getTopic(), null,
+				null, key, om.writeValueAsString(input.getValue()), headers);
+
+		Future send = producer.send(rec);
+		producer.close();
+
+		long timeTakenToPublish = Instant.now().toEpochMilli() - startPublishingEpochMillis;
+		logger.info("Time taken publishing {}", timeTakenToPublish);
```

Comment:
Consider using Spectator to collect durations as metrics.

Reply:
I see that on updateTask we are already pushing execution metrics to the spectator. I am thinking of not adding spectator here.